example/rcnn/rcnn/detector.py

```diff
@@ -33,36 +33,50 @@ def im_detect(self, im_array, im_info=None, roi_array=None):
+            # fill in data
+            if config.TEST.HAS_RPN:
+                self.arg_params['data'] = mx.nd.array(im_array, self.ctx)
+                self.arg_params['im_info'] = mx.nd.array(im_info, self.ctx)
+                arg_shapes, out_shapes, aux_shapes = \
+                    self.symbol.infer_shape(data=self.arg_params['data'].shape, im_info=self.arg_params['im_info'].shape)
+            else:
+                self.arg_params['data'] = mx.nd.array(im_array, self.ctx)
+                self.arg_params['rois'] = mx.nd.array(roi_array, self.ctx)
+                arg_shapes, out_shapes, aux_shapes = \
+                    self.symbol.infer_shape(data=self.arg_params['data'].shape, rois=self.arg_params['rois'].shape)
 
-        # fill in data
-        if config.TEST.HAS_RPN:
-            self.arg_params['data'] = mx.nd.array(im_array, self.ctx)
-            self.arg_params['im_info'] = mx.nd.array(im_info, self.ctx)
-            arg_shapes, out_shapes, aux_shapes = \
-                self.symbol.infer_shape(data=self.arg_params['data'].shape, im_info=self.arg_params['im_info'].shape)
-        else:
-            self.arg_params['data'] = mx.nd.array(im_array, self.ctx)
-            self.arg_params['rois'] = mx.nd.array(roi_array, self.ctx)
-            arg_shapes, out_shapes, aux_shapes = \
-                self.symbol.infer_shape(data=self.arg_params['data'].shape, rois=self.arg_params['rois'].shape)
+            # fill in label and aux
+            arg_shapes_dict = {name: shape for name, shape in zip(self.symbol.list_arguments(), arg_shapes)}
+            self.arg_params['cls_prob_label'] = mx.nd.zeros(arg_shapes_dict['cls_prob_label'], self.ctx)
+            aux_names = self.symbol.list_auxiliary_states()
+            self.aux_params = {k: mx.nd.zeros(s, self.ctx) for k, s in zip(aux_names, aux_shapes)}
 
-        # fill in label and aux
-        arg_shapes_dict = {name: shape for name, shape in zip(self.symbol.list_arguments(), arg_shapes)}
-        self.arg_params['cls_prob_label'] = mx.nd.zeros(arg_shapes_dict['cls_prob_label'], self.ctx)
-        aux_names = self.symbol.list_auxiliary_states()
-        self.aux_params = {k: mx.nd.zeros(s, self.ctx) for k, s in zip(aux_names, aux_shapes)}
-
-        # execute
-        self.executor = self.symbol.bind(self.ctx, self.arg_params, args_grad=None,
-                                         grad_req='null', aux_states=self.aux_params)
-        output_dict = {name: nd for name, nd in zip(self.symbol.list_outputs(), self.executor.outputs)}
-        self.executor.forward(is_train=False)
+            # execute
+            self.executor = self.symbol.bind(self.ctx, self.arg_params, args_grad=None,
+                                             grad_req='null', aux_states=self.aux_params)
+            executor = self.executor
+        else:
+            # Test whether we need to upsizing
+            if np.prod(im_array.shape) > np.prod(self.executor.arg_dict['data'].shape):
```

Comment:
@precedenceguo Also, when running the original test script, I find the memory consumption will vary dramatically between around 2G to 7G and may occasionally cause OOM. So I make this change. Have you met this problem before?